include/mender-deployment-data.h

```diff
@@ -78,7 +81,7 @@ mender_err_t mender_get_deployment_data(mender_deployment_data_t **deployment_da
  * @return MENDER_OK on success, otherwise MENDER_FAIL
  */
 mender_err_t mender_create_deployment_data(
-    const char *id, const char *artifact_name, const char *provides, const char *name, mender_deployment_data_t **deployment_data);
+    const char *id, const char *artifact_name, const char *provides, const char *state, mender_deployment_data_t **deployment_data);
```

Comment:
Pre-existing,

Because we are actually passing `NULL`, it "works". But this should instead be
```suggestion
    const char *id, const char *artifact_name, const char *provides, const mender_update_state_t *state, mender_deployment_data_t **deployment_data);
```

What is more, if we are only passing NULL, shouldn't we just remove it from the arguments? (same for provides) Or is it value in leaving it for some reason that I cannot see?

Reply:
I believe we can remove them from the arguments, unless you added them there for a specific reason @larsewi

Reply:
Yes, no issue in removing them ;) But should add a comment in the doc string that these must be populated later